packages/block-editor/src/components/block-types-list/index.js

```diff
@@ -13,45 +8,47 @@ import { getBlockMenuDefaultClassName } from '@wordpress/blocks';
  */
 import InserterListItem from '../inserter-list-item';
 
-function BlockTypesList( { items, onSelect, onHover = () => {}, children } ) {
+function BlockTypesList( { items = [], onSelect, onHover = () => {}, children } ) {
+	const normalizedItems = items.reduce( ( result, item ) => {
+		const { patterns = [] } = item;
+		const hasDefaultPattern = patterns.filter( ( { isDefault } ) => isDefault ).length > 0;
+
+		// If there is no default inserter pattern provided,
+		// then default block type is displayed.
+		if ( ! hasDefaultPattern ) {
+			result.push( item );
+		}
+
+		if ( patterns.length ) {
+			result.push( ...patterns.map( ( pattern ) => {
```

Comment:
I don't expect us to have a million elements in `patterns`, but maybe it'd be safer to use `concat` instead of spreading the arguments of `push`?

Reply:
good point 👍

Reply:
Addressed in 721026e.